File: apps/android/app/src/main/java/com/nutri/android/domain/FotoEscala.kt
package com.nutri.android.domain

object FotoEscala {
    const val LADO_MAXIMO = 1280
    const val QUALIDADE_JPEG = 70

    fun dimensoes(largura: Int, altura: Int, maxLado: Int = LADO_MAXIMO): Pair<Int, Int> {
        val maior = maxOf(largura, altura)
        if (maior <= maxLado || maior == 0) return largura to altura
        val fator = maxLado.toDouble() / maior.toDouble()
        val w = (largura * fator).toInt().coerceAtLeast(1)
        val h = (altura * fator).toInt().coerceAtLeast(1)
        return w to h
    }
}

File: apps/rn/src/domain/fotoEscala.ts
export const LADO_MAXIMO = 1280;
export const QUALIDADE_JPEG = 70;

export function dimensoesFoto(
  largura: number,
  altura: number,
  maxLado: number = LADO_MAXIMO,
): { largura: number; altura: number } {
  const maior = Math.max(largura, altura);
  if (maior <= maxLado || maior === 0) return { largura, altura };
  const fator = maxLado / maior;
  return {
    largura: Math.max(1, Math.trunc(largura * fator)),
    altura: Math.max(1, Math.trunc(altura * fator)),
  };
}
